<?xml version="1.0"?>

<launch>

  <!-- manager -->
  <arg name="use_nodelet" default="false" />
  <arg name="start_manager" default="false" />
  <arg name="manager" default="standalone_nodelet" />

  <!-- example node/nodelet parameters -->
  <arg name="topic_sub_image" default="/camera/image" />
  <arg name="topic_sub_camera_info" default="/topic_sub_camera_info" />
  <arg name="topic_pub_image" default="/topic_pub_image" />


  <group ns="/example_namespace">

    <node pkg="ros_node_nodelet_example" type="ros_node_nodelet_example_main" name="ExampleClassNode" unless="$(arg use_nodelet)" output="screen">
      <param name="topic_sub_image" value="$(arg topic_sub_image)" />
      <param name="topic_sub_camera_info" value="$(arg topic_sub_camera_info)" />
      <param name="topic_pub_image" value="$(arg topic_pub_image)" />
    </node>

    <node pkg="nodelet" type="nodelet" name="$(arg manager)" args="manager" if="$(arg start_manager)" output="screen" />

    <node pkg="nodelet" type="nodelet" name="ExampleClassNodelet" args="load ros_node_nodelet_example/ExampleClassNodelet $(arg manager)" if="$(arg use_nodelet)" output="screen">
      <param name="topic_sub_image" value="$(arg topic_sub_image)" />
      <param name="topic_sub_camera_info" value="$(arg topic_sub_camera_info)" />
      <param name="topic_pub_image" value="$(arg topic_pub_image)" />
    </node>

  </group>
</launch>
